src/test/java/com/artipie/FilesRepoITCase.java

```diff
@@ -0,0 +1,163 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie;
+
+import com.amihaiemil.eoyaml.Yaml;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.blocking.BlockingStorage;
+import com.artipie.asto.fs.FileStorage;
+import com.artipie.asto.test.TestResource;
+import com.jcabi.log.Logger;
+import java.nio.file.Path;
+import java.util.regex.Pattern;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.StringContains;
+import org.hamcrest.text.MatchesPattern;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.io.TempDir;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+
+/**
+ * Integration tests for Files repository.
+ * @since 0.11
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+@EnabledOnOs({OS.LINUX, OS.MAC})
+final class FilesRepoITCase {
+
+    /**
+     * Temporary directory for all tests.
+     * @checkstyle VisibilityModifierCheck (3 lines)
+     */
+    @TempDir
+    Path tmp;
+
+    /**
+     * Tested Artipie server.
+     */
+    private ArtipieServer server;
+
+    /**
+     * Container.
+     */
+    private GenericContainer<?> cntn;
+
+    /**
+     * Storage.
+     */
+    private Storage storage;
+
+    /**
+     * Artipie server port.
+     */
+    private int port;
+
+    @BeforeEach
+    void init() throws Exception {
+        this.storage = new FileStorage(this.tmp);
+        this.server = new ArtipieServer(this.tmp, "my-file", this.configs());
+        this.port = this.server.start();
+        this.server.start();
+        Testcontainers.exposeHostPorts(this.port);
+        this.cntn = new GenericContainer<>("centos:centos8")
+            .withCommand("tail", "-f", "/dev/null")
+            .withWorkingDirectory("/home/")
+            .withFileSystemBind(this.tmp.toString(), "/home");
+        this.cntn.start();
+        this.exec("yum", "-y", "install", "curl");
+        this.exec("yum", "-y", "install", "maven");
+    }
+
+    @Test
+    void downloadsArtifact() throws Exception {
+        final String url = "http://host.testcontainers.internal:%d/my-file/helloworld-src/pom.xml";
+        this.addFilesToStorage(
+            "helloworld-src", new Key.From("my-file", "helloworld-src")
+        );
+        MatcherAssert.assertThat(
+            "curl PUT doesn't work properly",
```

Comment:
@genryxy it's common to write what we expect, not what what happend on a failure. Meaning this would be better change to `curl PUT does work properly`. Same goes for other assert comment about `curl GET`